### PR TITLE
update bedrock contract deployment

### PR DESCRIPTION
**Pull request by dapperscene6**
Adds a few additional checks and updates bedrock configuration to match our current deploy plan

### CODE REVIEW

File: script/deploy/l1/DeployBedrock.s.sol [hunk 1]
```diff
@@ -330,15 +331,19 @@ contract DeployBedrock is Script {
         require(addressManager.owner() == address(proxyAdmin), "Deploy: addressManager owner is incorrect");
         vm.prank(address(0));
         require(l1StandardBridgeProxy.getOwner() == address(proxyAdmin), "Deploy: l1StandardBridgeProxy owner is incorrect");
+        vm.prank(address(0));
+        require(l1ERC721BridgeProxy.admin() == address(proxyAdmin), "Deploy: l1ERC721BridgeProxy owner is incorrect");
 
         SystemDictator.L2OutputOracleDynamicConfig memory l2OutputOracleDynamicConfig;
         l2OutputOracleDynamicConfig.l2OutputOracleStartingBlockNumber = deployConfig.l2OutputOracleStartingBlockNumber;
-        l2OutputOracleDynamicConfig.l2OutputOracleStartingTimestamp = block.timestamp;
+        l2OutputOracleDynamicConfig.l2OutputOracleStartingTimestamp = deployConfig.l2OutputOracleStartingBlockNumber;
```
Comment (dapperscene6) [inline script/deploy/l1/DeployBedrock.s.sol]: Change to `deployConfig.l2OutputOracleStartingTimestamp`

File: script/deploy/l1/DeployBedrock.s.sol [hunk 2]
```diff
@@ -202,6 +202,7 @@ contract DeployBedrock is Script {
             deployConfig.p2pSequencerAddress,
             defaultResourceCfg
         );
+        require(deployConfig.l2GenesisBlockGasLimit > defaultResourceCfg.systemTxMaxGas + defaultResourceCfg.maxResourceLimit, "Deploy: l2GenesisBlockGasLimit too low");
```
Comment (mdehoog) [inline script/deploy/l1/DeployBedrock.s.sol]: `>=`

File: script/deploy/l1/DeployBedrock.s.sol [hunk 3]
```diff
@@ -347,17 +353,19 @@ contract DeployBedrock is Script {
...
+        require(prevBaseFee == 1 gwei, "Deploy: optimismPortalProxy prevBaseFee is incorrect");
+        require(prevBoughtGas == 0, "Deploy: optimismPortalProxy prevBoughtGas is incorrect");
+        require(prevBlockNum != 0, "Deploy: optimismPortalProxy prevBlockNum is incorrect");
+        require(address(l1StandardBridgeProxy).balance == 0, "Deploy: l1StandardBridgeProxy balance is incorrect");
         require(address(L1StandardBridge(payable(address(l1StandardBridgeProxy))).messenger()) == address(l1CrossDomainMessengerProxy), "l1StandardBridgeProxy messenger is incorrect");
         require(OptimismMintableERC20Factory(address(optimismMintableERC20FactoryProxy)).BRIDGE() == address(l1StandardBridgeProxy), "optimismMintableERC20FactoryProxy l1StandardBridgeProxy is incorrect");
         require(address(L1ERC721Bridge(address(l1ERC721BridgeProxy)).messenger())== address(l1CrossDomainMessengerProxy), "l1ERC721BridgeProxy messenger is incorrect");
 
         vm.broadcast(deployer);
         systemDictator.finalize();
```
Comment (anikaraghu) [inline script/deploy/l1/DeployBedrock.s.sol:381]: require `systemDictator.finalized()`? https://github.com/ethereum-optimism/optimism/blob/3580bf1b41d80fcb2b895d5610836bfad27fc989/packages/contracts-bedrock/deploy/021-SystemDictatorSteps-2.ts#L348-L356

File: script/deploy/l1/DeployBedrock.s.sol [hunk 4]
```diff
@@ -316,48 +321,66 @@ contract DeployBedrock is Script {
...
-        require(addressManager.owner() == address(proxyAdmin), "Deploy: addressManager owner is incorrect");
-        vm.prank(address(0));
-        require(l1StandardBridgeProxy.getOwner() == address(proxyAdmin), "Deploy: l1StandardBridgeProxy owner is incorrect");
-
         SystemDictator.L2OutputOracleDynamicConfig memory l2OutputOracleDynamicConfig;
         l2OutputOracleDynamicConfig.l2OutputOracleStartingBlockNumber = deployConfig.l2OutputOracleStartingBlockNumber;
-        l2OutputOracleDynamicConfig.l2OutputOracleStartingTimestamp = block.timestamp;
+        l2OutputOracleDynamicConfig.l2OutputOracleStartingTimestamp = deployConfig.l2OutputOracleStartingTimestamp;
+
+        bool optimismPortalDynamicConfig = true;
```
Comment (anikaraghu) [inline script/deploy/l1/DeployBedrock.s.sol:346]: tiny nit: could comment what this means

**Pull request merged**